airbyte-integrations/connectors/source-orb/source_orb/source.py

```diff
@@ -177,6 +180,261 @@ def transform_record(self, subscription_record):
         return subscription_record
 
 
+# helpers for working with pendulum dates and strings
+def to_datetime(time) -> pendulum.DateTime:
+    if time is None:
+        return None
+    elif isinstance(time, pendulum.DateTime):
+        return time
+    elif isinstance(time, str):
+        return pendulum.parse(time)
+    else:
+        raise TypeError(f"Cannot convert input of type {type(time)} to DateTime")
+
+
+def to_utc_isoformat(time) -> str:
+    if time is None:
+        return None
+    elif isinstance(time, pendulum.DateTime):
+        return time.in_timezone("UTC").isoformat()
+    elif isinstance(time, str):
+        return pendulum.parse(time).in_timezone("UTC").isoformat()
+    else:
+        raise TypeError(f"Cannot convert input of type {type(time)} to isoformat")
+
+def chunk_date_range(start_date: pendulum.DateTime, end_date: Optional[pendulum.DateTime] = None) -> Iterable[pendulum.Period]:
+    """
+    Yields a list of the beginning and ending timestamps of each day between the start date and now.
+    The return value is a pendulum.period
+    """
+    one_day = pendulum.duration(days=1)
+    end_date = end_date or pendulum.now()
+
+    # Each stream_slice contains the beginning and ending timestamp for a 24 hour period
+    chunk_start_date = start_date
+    while chunk_start_date < end_date:
+        chunk_end_date = min(chunk_start_date + one_day, end_date)
+        yield pendulum.period(chunk_start_date, chunk_end_date)
+        chunk_start_date = chunk_end_date
```

Comment:
the function needs to `yield from []`. Otherwise it'll return `None` if `chunk_start_date >= end_date`